components/rendering/src/markdown.rs

```diff
@@ -283,3 +283,25 @@ pub fn markdown_to_html(content: &str, context: &RenderContext) -> Result<Render
         })
     }
 }
+
+fn check_external_links(links: &[String]) -> Result<()> {
+    let mut err_msg = links.par_iter().filter_map(|link| {
```

Comment:
does the parallelism really help here? It is already rendering markdown in parallel so I'm not sure this has a big impact

Reply:
It appears so. If it encounters multiple links on a page, it otherwise goes through them one-by-one, most of time waiting for IO. The speedup comes from waiting on multiple requests to return at the same time.

Reply:
(for clarity, parallelism was previously only a page level, this adds parallelism to a link level)